pkg/apparchive/helm-v1beta2.go

```diff
@@ -0,0 +1,299 @@
+	ProcessImageOptions image.ProcessImageOptions
+}
+
+// WriteRenderedV1Beta2HelmCharts writes the rendered v1beta2 helm charts to the rendered directory for diffing
+func WriteRenderedV1Beta2HelmCharts(opts WriteRenderedV1Beta2HelmChartsOptions) error {
+	if opts.KotsKinds == nil || opts.KotsKinds.V1Beta2HelmCharts == nil {
+		return nil
+	}
+
+	for _, downstream := range opts.Downstreams {
+		for _, helmChart := range opts.KotsKinds.V1Beta2HelmCharts.Items {
+			// template the chart with the values to the rendered dir for the downstream
+			renderedPath := path.Join(opts.RenderedDir, downstream, "helm", helmChart.GetDirName())
+			chartDir := path.Join(opts.HelmDir, helmChart.GetDirName())
+			valuesPath := path.Join(chartDir, "values.yaml")
+			if err := templateV1Beta2HelmChartWithValuesToDir(&helmChart, chartDir, valuesPath, renderedPath, opts.Log.Debug); err != nil {
+				return errors.Wrap(err, "failed to template helm chart for rendered dir")
+			}
+		}
+	}
+
+	return nil
+}
+
+func templateV1Beta2HelmChartWithValuesToDir(helmChart *kotsv1beta2.HelmChart, chartDir, valuesPath, outputDir string, log func(string, ...interface{})) error {
+	cfg := &action.Configuration{
+		Log: log,
+	}
+	client := action.NewInstall(cfg)
+	client.DryRun = true
+	client.ReleaseName = helmChart.GetReleaseName()
+	client.Replace = true
+	client.ClientOnly = true
+	client.IncludeCRDs = true
+
+	client.Namespace = helmChart.Spec.Namespace
+	if client.Namespace == "" {
+		client.Namespace = util.PodNamespace
+	}
+
+	chartPath := path.Join(chartDir, fmt.Sprintf("%s-%s.tgz", helmChart.Spec.Chart.Name, helmChart.Spec.Chart.ChartVersion))
+
+	chartRequested, err := loader.Load(chartPath)
+	if err != nil {
+		return errors.Wrap(err, "failed to load chart")
+	}
+
+	if req := chartRequested.Metadata.Dependencies; req != nil {
+		if err := action.CheckDependencies(chartRequested, req); err != nil {
+			return errors.Wrap(err, "failed dependency check")
+		}
+	}
+
+	values, err := chartutil.ReadValuesFile(valuesPath)
+	if err != nil {
+		return errors.Wrap(err, "failed to read values file")
+	}
+
+	rel, err := client.Run(chartRequested, values)
+	if err != nil {
+		return errors.Wrap(err, "failed to run helm install")
+	}
+
+	var manifests bytes.Buffer
+	fmt.Fprintln(&manifests, strings.TrimSpace(rel.Manifest))
+	for _, m := range rel.Hooks {
+		fmt.Fprintf(&manifests, "---\n# Source: %s\n%s\n", m.Path, m.Manifest)
+	}
+	// add hooks
+	for _, m := range rel.Hooks {
+		fmt.Fprintf(&manifests, "---\n# Source: %s\n%s\n", m.Path, m.Manifest)
+	}
+
+	if err := os.MkdirAll(outputDir, 0744); err != nil {
+		return errors.Wrap(err, "failed to create rendered path")
+	}
+
+	err = os.WriteFile(filepath.Join(outputDir, "all.yaml"), manifests.Bytes(), 0644)
+	if err != nil {
+		return errors.Wrap(err, "failed to write all.yaml")
+	}
+
+	return nil
+}
+
+func processOnlineImages(opts WriteV1Beta2HelmChartsOptions, helmChart *kotsv1beta2.HelmChart, chartDir string) error {
```

Comment:
since this is a shared package, let's name functions specifically. how about:
```suggestion
func processV1Beta2HelmChartImages(opts WriteV1Beta2HelmChartsOptions, helmChart *kotsv1beta2.HelmChart, chartDir string) error {
```